app/plex_tools.py

```diff
@@ -354,55 +431,55 @@ def delete_collection(data):
-    connection = sqlite3.connect(cache)
+    cache = "{}.cache".format(os.path.splitext(config_path)[0])
     with sqlite3.connect(cache) as connection:
         connection.row_factory = sqlite3.Row
         cursor = connection.cursor()
         cursor.execute(''' SELECT count(name) FROM sqlite_master WHERE type='table' AND name='guids' ''')
         if cursor.fetchone()[0] != 1:
-            print("| Initializing cache database.".format(cache))
-            cursor.execute('CREATE TABLE IF NOT EXISTS guids (plex_guid TEXT PRIMARY KEY, imdb_id TEXT, tmdb_id TEXT)')
+            print("| Initializing cache database at {}".format(cache))
+            cursor.execute('CREATE TABLE IF NOT EXISTS guids (plex_guid TEXT PRIMARY KEY, id TEXT, updated TEXT)')
+        else:
+            print("| Using cache database at {}".format(cache))
 
-def query_cache(config_path, key, column):
-    cache = os.path.join(os.path.dirname(config_path), 'cache.db')
+def query_cache(config_path, key):
+    cache = "{}.cache".format(os.path.splitext(config_path)[0])
+    id_to_return = None
+    update = None
     with sqlite3.connect(cache) as connection:
         connection.row_factory = sqlite3.Row
         cursor = connection.cursor()
         cursor.execute("SELECT * FROM guids WHERE plex_guid = ?", (key, ))
         row = cursor.fetchone()
-        if row:
-            return row[column]
+        if row and row["id"]:
+            datetime_object = datetime.strptime(row["updated"], "%Y-%m-%d")
+            time_between_insertion = datetime.now() - datetime_object
+            id_to_return = int(row["id"])
+            update = True if time_between_insertion.days > 30 else False
+    return id_to_return, update
 
-def update_cache(config_path, plex_guid, **kwargs):
-    cache = os.path.join(os.path.dirname(config_path), 'cache.db')
+def update_cache(config_path, plex_guid, input_id):
+    updated_date = datetime.now() - timedelta(days=random.randint(1,10))
+    cache = "{}.cache".format(os.path.splitext(config_path)[0])
     with sqlite3.connect(cache) as connection:
         connection.row_factory = sqlite3.Row
         cursor = connection.cursor()
-        if 'imdb_id' in kwargs:
-            imdb_id = kwargs['imdb_id']
-            cursor.execute('INSERT OR IGNORE INTO guids(plex_guid, imdb_id) VALUES(?, ?)', (plex_guid, imdb_id, ))
-            cursor.execute('UPDATE guids SET imdb_id = ? WHERE plex_guid = ?', (imdb_id, plex_guid))
-        if 'tmdb_id' in kwargs:
-            tmdb_id = kwargs['tmdb_id']
-            cursor.execute('INSERT OR IGNORE INTO guids(plex_guid, tmdb_id) VALUES(?, ?)', (plex_guid, tmdb_id, ))
-            cursor.execute('UPDATE guids SET tmdb_id = ? WHERE plex_guid = ?', (tmdb_id, plex_guid))
+        cursor.execute('INSERT OR IGNORE INTO guids(plex_guid) VALUES(?)', (plex_guid, ))
+        cursor.execute('UPDATE guids SET id = ?, updated = ? WHERE plex_guid = ?', (input_id, updated_date.strftime("%Y-%m-%d"), plex_guid))
```

Comment:
Why combine the id column? There's value in saving both imdb and tmdb ids, and it's a minimal operation.

Reply:
its minimal to add the ids but if we have to get the tmdb id and imdb id for every movie that could add alot of extra api calls for people using the old agent or for show agents which store it as tvdb id in the id column

Reply:
It wouldn't be any extra API calls because it's requesting the XML one time, then iterating through the guids. I really think all the IDs should be preserved here. It would benefit if someone were to switch agents. I see that you're storing an update timestamp and key, which I agree should help cut down the api hits for the old agent. One thing to consider is that, if you ignore the plex key, the cache file could actually be shared between users--even more useful if all the IDs are preserved.

Reply:
Yes but if you're on a different agent like the tmdb agent you'd have to make an extra api call per movie to store both tmdb and IMDb ids and tbh I don't see a huge benefit to storing both. the whole program is based on using tmdb ids, tvdb ids, or the plex rating key.

Reply:
If you are using the TMDB agent, then just store the TMDB ID and rkey. I agree there's no need for the IMDB ID then. But if the GUID indicates the item is using the Plex agent, then store TMDB ID, IMDB ID, and rkey. There's really no drawback here.

Reply:
Every time the program runs it will have to ping an api for every IMDb id to convert it to tmdb id vs just using a stored tmdb id so there's definitely implications of using both ids and I don't see what storing the IMDb Id would even do we're not going to use it

Reply:
thinking more on this yes we could store both but i would always store tmdb ID as a primary ID for movies and tvdb ID as a primary ID for shows If you really want to keep imdb i would suggest we instead use primary ID and alternative ID, but I'm of the opinion that we have absolutely no need to store the imdb ID its just never going to be used by the script so why would we bother using it.